### PR TITLE
#101 - Error Toast/Error Banner Example

### LINKED ISSUE #101: General - Add Toasts and Error Banners for Better Messaging UX
### Goal

Improve UX of error messaging by adding either error banners or toasts to accomplish messaging errors without blocking the whole page.

### Acceptance Criteria

show an example of this working with the MUI alert stuff so we can get UI/UX review on it before doing it everywhere

it should essentially replace one of our `alert(error)` calls

https://mui.com/material-ui/react-snackbar/

https://mui.com/material-ui/react-alert/

### PR DESCRIPTION
**Pull request by Ari-S-123**
## Changes

Added example of error toast/error banner.

## Screenshots

Screenshot 2023-01-17 154310

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #101

### CODE REVIEW

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice! i like it a lot visually. I think to avoid duplicated code though we need some kind of custom component and maybe even a custom hook for it

File: src/frontend/src/pages/ChangeRequestDetailPage/ReviewChangeRequestView.tsx [hunk 1]
```diff
@@ -69,89 +71,99 @@ const ReviewChangeRequestsView: React.FC<ReviewChangeRequestViewProps> = ({
...
-            '&::-webkit-scrollbar': {
-              display: 'none'
-            }
-          }}
-        >
-          <Typography sx={{ paddingBottom: 1 }}>{'Select Proposed Solution'}</Typography>
-          <Box
+      <>
+        <Toast open={showAlert} severity="error" message={'Please select a proposed solution!'} stateSetter={setShowAlert} />
+        <Dialog open={modalShow} onClose={onHide} style={{ color: 'black' }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ReviewChangeRequestView.tsx:76]: Could you make the Toast duration a lil longer, also just so this PR demonstrates how to use the duration prop

File: src/frontend/src/components/Toast.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Snackbar, Alert } from '@mui/material';
+
+interface ToastProps {
+  message: string;
+  autoHideDuration?: number;
```
Comment (RChandler234) [inline src/frontend/src/components/Toast.tsx:10]: add a comment that this is in milliseconds

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — DISMISSED** [review]

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
the issue with this implementation is that if we want two alerts on the same page, they'll be in separate snackbars which means they will overlay each other. not sure the best way to fix this... maybe separating snackbar and alert into custom stuff? or some way to have a list of alerts within the Toast component?

what if it's like

```
<Toast
...
alerts={[ { message: 'a', type: 'error', open: showAlert1, handleClose: () => setShowAlert1(false) } ]}
/>
```

so the prop is of that type and that? I have no idea if this is a good idea and i wonder if it's something react will get mad about. i feel like there has to be an easier way to do this and im just way overthinking it

**Review by anthonybernardi — APPROVED** [review]
very clean

**Pull request merged**